piptools/scripts/compile.py

```diff
@@ -42,10 +45,44 @@
     "pyproject.toml",
     "setup.cfg",
 )
+ALL_BUILD_DISTRIBUTIONS: tuple[Literal["sdist", "wheel", "editable"], ...] = (
+    "editable",
+    "sdist",
+    "wheel",
+)
+DEFAULT_REQUIREMENTS_FILE = "requirements.in"
 DEFAULT_REQUIREMENTS_OUTPUT_FILE = "requirements.txt"
 METADATA_FILENAMES = frozenset({"setup.py", "setup.cfg", "pyproject.toml"})
 
 
+def _build_requirements(
+    src_dir: str, src_file: str, distributions: Iterable[str], package_name: str
+) -> list[InstallRequirement]:
+    assert not isinstance(distributions, str)
+    result = collections.defaultdict(set)
+
+    with IsolatedEnvBuilder() as env:
+        builder = ProjectBuilder(
+            srcdir=src_dir,
+            python_executable=env.executable,
+            scripts_dir=env.scripts_dir,
+            runner=pyproject_hooks.quiet_subprocess_runner,
```

Comment:
```suggestion
            runner=pyproject_hooks.default_subprocess_runner,
```

I'd suggest using `default_subprocess_runner` instead. The `quiet_subprocess_runner` suppresses any error messages on the build stage. Please see the following links for details:
- https://github.com/jazzband/pip-tools/issues/1711#issuecomment-1328301486
- https://github.com/pypa/build/issues/553

Reply:
Should this be verbosity-dependent, then?

Reply:
Makes perfect sense!

Reply:
Done.

Reply:
Thanks!